scrapy/extensions/feedexport.py

```diff
@@ -288,8 +294,7 @@ def close_spider(self, spider):
     def item_scraped(self, item, spider):
         for slot in self.slots:
             slot.start_exporting()
```

Comment:
I think this may start export without a need, if an item would be filtered out. 

Let's say you have a single FEEDS configuration in settings.py, which defines 2 export locations, split by 2 item classes. There are 2 spiders; one spider emits items of type A, the other emits items of type B. While running a spider, empty (content-wise) file can be created for the item type which spider doesn't emit.

Reply:
I don’t see how we can avoid this. I think in this scenario `FEEDS` would need to be redefined on each spider, if creating an empty file is an issue.

Reply:
Inside `slot.export_item()` (called in a next line) there is a `if self.exporter._should_export_item(item):` check. My point was that it could make sense to call `slot.start_exporting()` only after this check is successful at least once. It won't fix all scenarios where an empty file can appear, but, if I'm not mistaken, this should fix the problem I described.